src/main/daml/Daml/Finance/Instrument/Generic/Util.daml

```diff
@@ -175,21 +178,79 @@ calculateClaimsFromFixSwapStream s periodicSchedule streamSchedule useAdjustedDa
   let f = fromSome s.calculationPeriodAmount.calculation.fixedRateSchedule
   pure $ createFixRatePaymentClaims streamSchedule periodicSchedule useAdjustedDatesForDcf f.initialValue issuerPaysLeg s.calculationPeriodAmount.calculation.dayCountFraction currency
 
+-- | Rule when one specific floating rate is provided for a stub period.
+getSingleStubRateFromFpml floatingRate = do
+  Some (Observe floatingRate.floatingRateIndex)
+
+-- | Rule when two specific floating rates are provided for a stub period.
+-- In this case, linear interpolation should be used, as specified here:
+-- https://www.isda.org/a/aWkgE/Linear-interpolation-04022022.pdf
+getInterpolatedStubRateFromFpml fr1 fr2 p cal convention = do
+  let
+    tenor1 = fromSome fr1.indexTenor
+    tenor2 = fromSome fr2.indexTenor
+    s = p.adjustedStartDate
+    p1 = adjustDate cal convention (addPeriod s tenor1)
+    p2 = adjustDate cal convention (addPeriod s tenor2)
+    tn = subDate p.adjustedEndDate s
+    t1 = subDate p1 s
+    t2 = subDate p2 s
+    d1 = tn - t1
+    d2 = t2 - tn
+    w1 = intToDecimal d2 / intToDecimal (d1 + d2)
```

Comment:
one `intToDecimal` is sufficient, no?

Reply:
I tried removing the first one and the second one, respectively (did not work).
Also, tried the following:
`    w1 = intToDecimal (d2 / (d1 + d2))
`but it gives a different result.

Reply:
ok, keep as is. the `w1 = intToDecimal (d2 / (d1 + d2))` is doing integer division.